lib/ansibledocgen/parser/playbook.py

```diff
@@ -106,17 +107,20 @@ def parse_playbook(self, playbook):
             if yamldata is None:
                 return
 
-            # Parase the Yaml
+            # Parse the Yaml
+            task_names = []
             for yaml_item in yamldata:
-                tasks = yaml_item
                 # Playbooks have a tasks dict key
-                if "tasks" in yaml_item:
-                    tasks = yaml_item["tasks"]
-                # Loop through tasks
-                task_info = self.__get_task_info__(tasks)
-                if len(task_info) > 0:
-                    playbookentry["task_info"] += task_info
-                # Loop through Playbook tasks
+                if "name" in yaml_item:
+                    task_names.append(yaml_item["name"])
+                if "roles" in yaml_item:
+                    roles = yaml_item["roles"]
+                    role_info = []
+                    for i in range(len(roles)):
+                        role_info.append(roles[i])
+                    playbookentry["role_info"] = role_info
+            playbookentry["task_names"] = task_names
+            # Loop through Playbook tasks
             if folder_content not in self.parserdata:
```

Comment:
There's a lot more to this than I expected.  Can we have a chat please, so I can understand what's going on?

Reply:
As discussed, please add some comments explaining what you just told me!